tests/ttsim/interface_dag_elements/test_failures.py

```diff
@@ -784,7 +784,7 @@ def test_fail_if_non_convertible_objects_in_results_tree_because_of_object_type(
     with pytest.raises(TypeError, match=match):
```

Comment:
Just wondering whether we should just re-write the tests that the error will be raised in the `main` call? Would seem clearer to me.